lib/screens/v2/candidate_generator/widgets/cr_departures.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Screens.V2.CandidateGenerator.Widgets.CRDepartures do
+  @moduledoc false
+
+  alias Screens.Config.Screen
+  alias Screens.Config.V2.{CRDepartures, PreFare}
+  alias Screens.Stops.Stop
+  alias Screens.V2.Departure
+  alias Screens.V2.WidgetInstance.CRDepartures, as: CRDeparturesWidget
+  alias Screens.V2.WidgetInstance.{DeparturesNoData, OvernightDepartures}
+
+  def departures_instances(
+        config,
+        fetch_departures_fn \\ &fetch_departures/2,
+        fetch_stop_name_fn \\ &Stop.fetch_stop_name/1
+      )
+
+  def departures_instances(
+        %Screen{app_params: %app{cr_departures: %CRDepartures{} = cr_departures}} = config,
+        fetch_departures_fn,
+        fetch_stop_name_fn
+      )
+      when app in [PreFare] do
+    opts = %{include_schedules: true}
+
+    params = %{
+      direction_id: cr_departures.direction_to_destination,
+      route_type: :rail,
+      stop_ids: [cr_departures.station]
+    }
+
+    destination = fetch_stop_name_fn.(cr_departures.destination)
+
+    {:ok, departures_data} = fetch_departures_fn.(opts, params)
+
+    # The Overnight and NoData widgets may not be relevant here
+    departures_instance =
+      cond do
+        departures_data == :error ->
```

Comment:
I think this would mean we got `{:ok, :error}` back from `fetch_departures_fn.(opts, params)` above, is that right?

It also does not look like `Departure.fetch/2` can return `{:ok, :overnight}` but I might be missing something.

A `case` expression might work best here, again I'm not 100% sure of the possible return values but it looks like we're expecting something like
```ex
case fetch_departures_fn.(opts, params) do
  :error ->
    %DeparturesNoData{screen: config, show_alternatives?: true}

  {:ok, :overnight} ->
    %OvernightDepartures{}

  _ ->
    %CRDeparturesWidget{
      config: cr_departures,
      departures_data: departures_data,
      destination: destination
    }
end
```